src/index.js

```diff
@@ -169,12 +167,11 @@ function shimBelongsTo(target) {
 
   shimmer.wrap(target, 'get', original => {
     return function batchedGetBelongsTo(instance, options = {}) {
-      // targetKeyIsPrimary already handled by sequelize (maps to findById)
-      if (this.targetKeyIsPrimary || Array.isArray(instance) || options.include || options.transaction) {
```

Comment:
Skip the original code even if the target key is the primary - Since we cannot rely on the target model being shimmed, which would make the findByIds slow